spark/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -1124,6 +1140,48 @@ class DeltaAnalysis(session: SparkSession)
     DeltaViewHelper.stripTempViewForMerge(plan, conf)
   }
 
+  /**
+   * Recursively casts maps in case it contains null types.
```

Comment:
"Recursively casts map data types in case the key/value type differs"

I'm not entirely sure why the struct method mentions `in case it contains null types` but I don't think that's accurate